test/e2e/testing-manifests/storage-csi/mock/csi-mock-proxy.yaml

```diff
@@ -0,0 +1,90 @@
+kind: StatefulSet
+apiVersion: apps/v1
+metadata:
+  name: csi-mockplugin
+spec:
+  selector:
+    matchLabels:
+      app: csi-mockplugin
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: csi-mockplugin
+    spec:
+      serviceAccountName: csi-mock
+      containers:
+        - name: csi-provisioner
+          image: k8s.gcr.io/sig-storage/csi-provisioner:v1.6.0
+          args:
+            - "--csi-address=$(ADDRESS)"
+            # Topology support is needed for the pod rescheduling test
+            # ("storage capacity" in csi_mock_volume.go).
+            - "--feature-gates=Topology=true"
+            - "-v=5"
+            - "--timeout=1m"
+          env:
+            - name: ADDRESS
+              value: /csi/csi.sock
+          volumeMounts:
+            - mountPath: /csi
+              name: socket-dir
+        - name: driver-registrar
+          # TODO: replace with official image that includes https://github.com/kubernetes-csi/node-driver-registrar/pull/124
+          image: pohly/csi-node-driver-registrar:v1.3.0
+          args:
+            - --v=5
+            - --csi-address=/csi/csi.sock
+            - --kubelet-registration-path=/var/lib/kubelet/plugins/csi-mock/csi.sock
+            - --timeout=1m
+          env:
+            - name: KUBE_NODE_NAME
+              valueFrom:
+                fieldRef:
+                  apiVersion: v1
+                  fieldPath: spec.nodeName
+          securityContext:
+            privileged: true
+          volumeMounts:
+          - mountPath: /csi
+            name: socket-dir
+          - mountPath: /registration
+            name: registration-dir
+        - name: mock
+          # TODO: replace with official image which includes https://github.com/kubernetes-csi/csi-test/pull/314
+          image: pohly/mock-driver:2020-12-04-1
```

Comment:
Does it make sense to build a separate csi-mock-proxy container (+github repo)? Because it does not share much with the mock driver.

Reply:
We could do that. But I'd prefer to avoid the overhead of managing a separate repo and image. As an intermediate solution would could keep the code in the csi-test repo and introduce a `csi-proxy` command and image, but even that IMHO doesn't offer much actual advantages besides "it's a bit cleaner".

Reply:
Maybe a crazy idea, can the node driver registrar register a `tcp://` socket of a CSI driver + can we deploy a generic TCP proxy (HA proxy?). We won't need any extra container on the nodes.

Reply:
It's not just kubelet, we also want to test with the actual sidecar containers, right? So we do need to run containers inside Kubernetes. The sidecars also could use TCP, but I doubt that this makes a difference: the proxy must establish a connection between *two* incoming TCP connection requests, one coming from kubelet or sidecar, the other coming from e2e.test outside of the cluster. I doubt that a generic proxy can do that.

It would be more natural to forward a listening port opened by the e2e.test binary into the cluster, but that is not supported by the port-forwarding feature of the apiserver, at least as far as I know - it just supports incoming connections, not outgoing ones.

Reply:
Yet another possibility: create a `csi-proxy` command and publish it in the same `mock-driver` image. Hmm, this might be even more work than adding a second image. A second image should already be supported by the csi-release-tools (albeit not tested, I think) whereas a second command per image isn't.

Reply:
Ack, let's keep it in the mock driver for now. We can always split it later  if it becomes too complicated to maintain a proxy and a driver in the same binary.

Reply:
> Ack, let's keep it in the mock driver for now.

Can you then approve https://github.com/kubernetes-csi/csi-test/pull/314? That's the implementation of that approach.